flytekit/tools/translator.py

```diff
@@ -164,6 +191,62 @@ def fn(settings: SerializationSettings) -> List[str]:
     return fn
 
 
+@lru_cache
+def display_ipython_warning(input: str) -> None:
+    # This is a warning that is only displayed once per python type
+    logger.debug(input)
+
+
+def _update_serialization_settings_for_ipython(
+    entity: FlyteLocalEntity,
+    serialization_settings: SerializationSettings,
+    options: Optional[Options] = None,
+) -> SerializationSettings:
+    # If the entity is not a PythonAutoContainerTask, we don't need to do anything, as only Tasks with container |
+    # user code in container need to be serialized as pickled objects.
+    if not isinstance(entity, (PythonAutoContainerTask, ArrayNodeMapTask)):
+        return serialization_settings
+
+    # Let's check if we are in an interactive environment like Jupyter notebook
+    if serialization_settings.interactive_mode_enabled is True:
+        # We are in an interactive environment, let's check if the task is a PythonFunctionTask and the task function
+        # is defined in the main module. If so, we will serialize the task as a pickled object and upload it to remote
+        # storage. The main module check is to ensure that the task function is not defined in a notebook cell.
+        if isinstance(entity, PythonFunctionTask):
+            # if not entity.task_function.__module__ == "__main__":
+            #     raise FlyteAssertion(
+            #         "Task function should be defined in the main module | jupyter cell for"
+            #         " interactive mode. Task function defined in imported modules is not supported."
+            #         f" Task function {entity.task_function.__name__} is defined in an imported module"
+            #     )
+            if entity.execution_mode == PythonFunctionTask.ExecutionBehavior.DYNAMIC:
+                raise FlyteAssertion(
+                    f"Dynamic tasks are not supported in interactive mode. {entity.name} is a dynamic task."
+                )
+
+        if options is None or options.file_uploader is None:
+            raise FlyteAssertion("To work interactively with Flyte, a code transporter/uploader should be configured.")
+
+        import gzip
+
+        import cloudpickle
+
+        from flytekit.configuration import FastSerializationSettings
+
+        display_ipython_warning("Jupyter notebook and interactive task support is still alpha.")
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            dest = pathlib.Path(tmp_dir, "pkl.gz")
+            with gzip.GzipFile(filename=dest, mode="wb", mtime=0) as gzipped:
+                cloudpickle.dump(entity, gzipped)
+            display_ipython_warning("Uploading Pickled representation of Task to remote storage...")
```

Comment:
Can you check the size of `dest` and throw an error if the pickle is too big? I'll start with ~150 MB.

(Not for this PR) A quick followup is to do this check while pickling.

Reply:
@thomasjpfan I just moved to UCSD recently, and I’ll be updating the code this weekend. Once I’m settled in, I’ll have a lot of time to focus on this. I can follow your timeline to push this feature after next week. Feel free to reach out if you’d like to have a 1-on-1.